commands/command_prune.go

```diff
@@ -521,6 +522,26 @@ func pruneTaskGetRetainedStashed(gitscanner *lfs.GitScanner, retainChan chan str
 	}
 }
 
+// Background task, must call waitg.Done() once at end
+func pruneTaskGetRetainedIndex(gitscanner *lfs.GitScanner, retainChan chan string, errorChan chan error, waitg *sync.WaitGroup, sem *semaphore.Weighted) {
+	defer waitg.Done()
+
+	err := gitscanner.ScanIndex("HEAD", func(p *lfs.WrappedPointer, err error) {
```

Comment:
We probably want to handle multiple indices, one for each worktree.  It might therefore be helpful to call this from `pruneTaskGetRetainedWorktree`.  If we don't do that, we might produce wrong results if I do `git add` in worktree A and then `git lfs prune` in another worktree.